### PR TITLE
Focus first invalid field onSubmit, closes #32

### LINKED ISSUE #32: Focus first invalid form field if form submission doesn't pass validation

### PR DESCRIPTION
**Pull request by orischwartz**

### CODE REVIEW

File: app/assets/javascripts/form8.js [hunk 1]
```diff
@@ -153,9 +153,11 @@
     validateSubmit: function() {
       var self = this;
 
-      return this.getRequiredQuestions().reduce(function(result, questionNumber) {
-        return self.validateRequiredQuestion(questionNumber, true) && result;
-      }, true);
+      var invalidQuestionNumbers = $.grep(this.getRequiredQuestions(), function(questionNumber, index){
```
Comment (shanear) [inline app/assets/javascripts/form8.js]: I'm preferential to using ES5 language methods over jQuery methods when possible, but I'm ok with this too. What are your thoughts?

File: app/assets/javascripts/form8.js [hunk 2]
```diff
@@ -183,10 +185,19 @@
 
     onSubmit: function() {
       this.fetchState();
-      var result = this.validateSubmit();
+      var invalidQuestionNumbers = this.validateSubmit();
```
Comment (shanear) [inline app/assets/javascripts/form8.js]: maybe change the name of `validateSubmit()` to `getInvalidQuestionNumbers()` for clarity?

File: app/assets/javascripts/form8.js [hunk 3]
```diff
@@ -183,10 +185,19 @@
...
       this.render();
 
-      return result;
+      // invalid, focus first invalid field
+      if ( invalidQuestionNumbers.length > 0 ) {
+        $question(invalidQuestionNumbers[0]).find("input, textarea, select").first().focus();
+        return false;
+      }
+
+      // valid
```
Comment (shanear) [inline app/assets/javascripts/form8.js]: `return invalidQuestionNumbers.length > 0;` would eliminate the need for an else statement here (reducing cyclomatic complexity).

File: app/assets/javascripts/form8.js (hunk 1 above)
Comment (orischwartz) [inline app/assets/javascripts/form8.js]: My reflex is to use jQuery to get the broadest browser-support possible. But if we're only committing to IE9+ support, I'm ok with using ES5 whenever possible, will change.

**Pull request merged**